source/common/upstream/BUILD

```diff
@@ -373,6 +373,40 @@ envoy_cc_library(
         ":static_cluster_lib",
         ":strict_dns_cluster_lib",
         ":upstream_includes",
+        ":transport_socket_matcher_lib",
+        "//include/envoy/event:dispatcher_interface",
+        "//include/envoy/event:timer_interface",
+        "//include/envoy/network:dns_interface",
+        "//include/envoy/network:listen_socket_interface",
+        "//include/envoy/ssl:context_interface",
+        "//include/envoy/upstream:health_checker_interface",
+        "//source/common/common:enum_to_int",
+        "//source/common/common:utility_lib",
+        "//source/common/config:protocol_json_lib",
+        "//source/common/http:utility_lib",
+        "//source/common/network:address_lib",
+        "//source/common/network:resolver_lib",
+        "//source/common/network:socket_option_factory_lib",
+        "//source/common/network:utility_lib",
+        "//source/common/protobuf",
+        "//source/common/protobuf:utility_lib",
+        "//source/extensions/transport_sockets:well_known_names",
+        "//source/server:transport_socket_config_lib",
+        "@envoy_api//envoy/api/v2/core:base_cc",
+    ],
+)
+
+envoy_cc_library(
+    name = "transport_socket_matcher_lib",
+    srcs = ["transport_socket_matcher.cc"],
+    deps = [
+        ":eds_lib",
```

Comment:
do you need all of them? Can you strip these down to what you need?

Reply:
done. removed unnecessary ones.